util/circular_buffer.c

```diff
@@ -146,12 +141,148 @@ int32_t cb_remove(struct circular_buffer *desc)
  *  - \ref SUCCESS : On success
  *  - \ref FAILURE : Otherwise
  */
-int32_t cb_size(struct circular_buffer *desc, uint32_t *nb_elements)
+int32_t cb_size(struct circular_buffer *desc, uint32_t *size)
+{
+	uint32_t overflows;
+	uint32_t overflowed_size;
+
+	if (!desc || !size)
+		return -EINVAL;
+
+	if (desc->write.nb_overflows < desc->read.nb_overflows)
+		overflows = UINT32_MAX - desc->read.nb_overflows +
+			    desc->write.nb_overflows;
+	else
+		overflows = desc->write.nb_overflows - desc->read.nb_overflows;
+
+	/* Min function is used to avoid an integer overflow. */
+	overflowed_size = desc->size * min(2, overflows);
```

Comment:
Which integer's overflow does the min function prevent?

Reply:
overflow * desc->size can overflwo. But there is no need to consider 2 overflows in order to get the correct size.

Reply:
II updated the cb_size implementation